fmf/base.py

```diff
@@ -606,6 +607,44 @@ def copy(self):
         self.parent = duplicate.parent = original_parent
         return duplicate
 
+    def validate(self, schema):
+        """
+        Validate node data with given JSON Schema.
+
+        Return a named tuple utils.JsonSchemaValidationResult
+        with the following two items:
+
+          result ... boolean representing the validation result
+          errors ... A list of validation errors
+
+        Raises utils.JsonSchemaError if the supplied schema was invalid.
+        """
+        try:
+            resolver = jsonschema.RefResolver.from_schema(schema)
+        except AttributeError as error:
+            raise utils.JsonSchemaError(
+                f'Provided schema cannot be loaded: {error}')
+
+        validator = jsonschema.Draft4Validator(schema, resolver=resolver)
+
+        try:
+            validator.validate(self.data)
+            return utils.JsonSchemaValidationResult(True, [])
+
+        # Data file validated by schema contains errors
+        except jsonschema.exceptions.ValidationError:
+            return utils.JsonSchemaValidationResult(
+                False, list(validator.iter_errors(self.data)))
```

Comment:
Shall we convert the generator to `list` to match the `List[Any]` type? Otherwise we get this:

    JsonSchemaValidationResult(result=False, errors=<generator object create.<locals>.Validator.iter_errors at 0x7f40dd8fdf50>)

Reply:
+1

Reply:
@psss Ack, I can fix it, but this file is not yet typed, these issues will appear once we add typing to fmf

Reply:
Good, already fixed in cec4c39. Going to squash and merge.

Reply:
right, you did it, nothing I can do here